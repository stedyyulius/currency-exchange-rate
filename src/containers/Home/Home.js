import React, { useState } from 'react';
import { connect } from 'react-redux';

import styles from './Home.module.css'

import ExchangeRatesBox from '../../components/ExchangeRatesBox/ExchangeRatesBox';

const Home = props => {
    const [isAddCurrency, setIsAddCurrency] = useState(false);

    const addCurrency = () => {
        setIsAddCurrency(true);
    }

    const addCurrencyButton = () => {
        if (isAddCurrency) {
            return (
                <div className={styles.addCurrencyButton} onClick={() => addCurrency()}>
                    <input type="text" />
                    <button>Submit</button>
                </div>
            )
        }

        return (
            <div className={styles.addCurrencyButton} onClick={() => addCurrency()}>
                <p>(+)</p>
                <p>Add More Currencies</p>
            </div>
        )
    }

    return (
        <div className={styles.container}>
            <div className={styles.rateBox}>
                <div className={styles.rateBoxTitle}>
                    <i>USD - United States Dollars</i>
                    <br />
                    <div className={styles.rateTitle}>
                        <b>USD</b>
                        <b>10.0000</b>
                    </div>
                </div>
                <div className={styles.rateBoxBody}>
                    <ExchangeRatesBox />
                </div>
                <div className={styles.addCurrencyContainer}>
                    {addCurrencyButton()}
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = state => {
    return {

    }
}

const mapDispatchToProps = dispatch => {
    return {

    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);