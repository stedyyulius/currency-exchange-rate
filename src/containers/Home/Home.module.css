.container {
  width: 100vw;
  height: 100vh;
  padding-top: 50px;
}

.rateBox {
  width: 500px;
  border: var(--themeBorder);
  margin: auto;
  padding-bottom: 20px;
}

.rateBoxTitle {
  padding: 15px;
  border-bottom: var(--themeBorder);
}

.rateTitle {
  display: flex;
  justify-content: space-between;
}

.rateBoxBody {
  padding: 15px;
  text-align: center;
}

.addCurrencyContainer {
  padding: 15px;
}

.addCurrencyButton {
  display: flex;
  margin: auto;
  border: var(--themeBorder);
  line-height: 0%;
}

.addCurrencyButton p {
  margin-left: 10px;
}

.addCurrencyButton input {
  width: 100%;
}
