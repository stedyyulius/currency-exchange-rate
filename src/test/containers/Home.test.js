import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { shallow, mount } from 'enzyme';

import Home from '../../containers/Home/Home';
import store from '../../store';

configure({ adapter: new Adapter() });

describe('Home', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<Home store={store} />).dive();
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain add currency button', () => {
        const wrapper = mount(<Home store={store} />);
        expect(wrapper.find('.addCurrencyButton').length).toEqual(1)
    });
});