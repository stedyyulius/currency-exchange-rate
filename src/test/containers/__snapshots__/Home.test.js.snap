// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<Home
  exchangeRates={Array []}
  getLatestRates={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
